Automattic/jetpack
# Social Icons Widget: Add more common feed url components to supported icons

As an improvement based on the issue #10662 "Social Icons Widget: allow for alternate feed URLs" this adds some more common feed url components to the get_supported_icons() function. The existing version only matches the existence of "/feed/" in the url being added, as such many other feed URL formats would end up showing with the default link icon instead. Whilst it may still not catch all possible feed url formats it will catch many commonly used formats as well as non-permalink versions of WordPress feeds.

This feature could be tested by beta testers to see if more feed urls are correctly recognised and shown with the RSS feed icon instead of the generic link icon.

Fixes #10662

#### Changes proposed in this Pull Request:
* This PR adds in approximately 20 additional items to an array `$social_links_icons` which is used to match components of any url added to the Social Icons widget, the additions expand upon the existing `/feed/` array item which is intended to show an RSS feed icon if matched. Whilst this works for the standard WordPress permalink feed format it doesn't work for non-permalink WordPress feed urls and also many other common RSS feed url formats.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This adds features to the `get_supported_icons()` function which matches urls added to the Social Icons widget to specific icons for the URL. This addition provides more items to the `$social_links_icons` array which match other common RSS feed url structures.

#### Testing instructions:
To test this please try adding additional RSS feed urls to the Social Icons widget, the standard WordPress url for `somedomain.com/feed/` is already supported so try some more unusual formats such as those generated by other CMS or blogging platforms. Here are some suggested urls to try:
* http://blogs.adobe.com/jkost/atom.xml
* http://feeds.feedburner.com/clickontyler
* http://daringfireball.net/index.xml
* http://ranchero.com/xml/rss.xml
* http://usefulmac.com/rss
* http://rss.macworld.com/macworld/feeds/main

If these are correctly parsed then you should see an RSS feed icon instead of the default 'chain link' icon.

#### Proposed changelog entry for your changes:
* This improves the likelihood that RSS Feed urls that are added to the Social Icons widget are correctly recognised and displayed using the RSS feed icon rather than the default link icon.
